disnake/ui/action_row.py

```diff
@@ -0,0 +1,251 @@
+
+        self._total_width += 1
+        self._underlying.children.append(
+            ButtonComponent._raw_construct(
+                type=ComponentType.button,
+                custom_id=custom_id,
+                url=url,
+                disabled=disabled,
+                label=label,
+                style=style,
+                emoji=emoji,
+            )
+        )
+
+    def add_select(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        options: List[SelectOption] = MISSING,
+        disabled: bool = False,
+    ):
+        """Adds a select menu to the action row.
+
+        To append a pre-existing :class:`disnake.ui.Select` use the
+        :meth:`append_item` method instead.
+
+        Parameters
+        -----------
+        custom_id: :class:`str`
+            The ID of the select menu that gets received during an interaction.
+            If not given then one is generated for you.
+        placeholder: Optional[:class:`str`]
+            The placeholder text that is shown if nothing is selected, if any.
+        min_values: :class:`int`
+            The minimum number of items that must be chosen for this select menu.
+            Defaults to 1 and must be between 1 and 25.
+        max_values: :class:`int`
+            The maximum number of items that must be chosen for this select menu.
+            Defaults to 1 and must be between 1 and 25.
+        options: List[:class:`disnake.SelectOption`]
+            A list of options that can be selected in this menu.
+        disabled: :class:`bool`
+            Whether the select is disabled or not.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width >= 5:
+            raise ValueError("Too many items in this row, can not add a select menu.")
+
+        self._total_width += 5
+        self._underlying.children.append(
+            SelectMenu._raw_construct(
+                custom_id=custom_id,
+                type=ComponentType.select,
+                placeholder=placeholder,
+                min_values=min_values,
+                max_values=max_values,
+                options=options,
+                disabled=disabled,
+            )
+        )
+
+    def to_component_dict(self) -> ActionRowPayload:
+        return self._underlying.to_dict()
+
+
+def components_to_dict(
+    components: Union[ActionRow, Item, List[Union[ActionRow, Item, List[Item]]]]
+) -> List[ActionRowPayload]:
+    if isinstance(components, ActionRow):
+        return [components.to_component_dict()]
+
+    if isinstance(components, Item):
+        return [ActionRow(components).to_component_dict()]
+
+    if not isinstance(components, list):
+        raise ValueError("components must be an Item, a list of ActionRows or a list of Items")
```

Comment:
This could maybe be replaced by just something like `if not isinstance(components, list): components = [components]`